### PR TITLE
Fix failing gouraud shading mask creation in pccolormesh

### PR DESCRIPTION
**Pull request by e-k-m**
## Rationale

Using pccolormesh with gouraud shading fails on wrap around mask creation, since numpy array mask is created with negative dimension. 

Details: Fails here https://github.com/SciTools/cartopy/blob/main/lib/cartopy/mpl/geoaxes.py#L1896 because 
data_shape equals (Nx, Ny, -1) or (Ny - 1, Nx - 1, -1). 

## Implications

As far as I can see there are no implications of the proposed change.

### CODE REVIEW

**Comment by e-k-m** [comment]
Docs generation seems to fail given an unrelated issue to this PR. Ready for review. THX.

File: lib/cartopy/crs.py
Comment (greglucas) [inline lib/cartopy/crs.py:1]: Could you revert the changes to this file? They are unrelated to your other updates.

File: lib/cartopy/tests/mpl/test_mpl_integration.py [hunk 1]
```diff
@@ -654,6 +654,21 @@ def test_pcolormesh_single_column_wrap():
...
+    x, y = np.meshgrid(x_range, y_range)
+    data = ((np.sin(np.deg2rad(x))) / 10. + np.exp(np.cos(np.deg2rad(y))))
+
+    fig = plt.figure(figsize=(10, 6))
+    ax = fig.add_subplot(1, 1, 1, projection=ccrs.Mercator())
+    try:
+        ax.pcolormesh(x, y, data, transform=ccrs.PlateCarree(), shading='gouraud')
+    except ValueError as e:
+        if str(e) == 'negative dimensions are not allowed':
+            pytest.fail("Unexpected ValueError('negative dimensions are not allowed')")
```
Comment (greglucas) [inline lib/cartopy/tests/mpl/test_mpl_integration.py:669]: pytest will fail for us, so no need to explicitly catch and fail.
```suggestion
    ax.pcolormesh(x, y, data, transform=ccrs.PlateCarree(), shading='gouraud')
```

**Review by greglucas — COMMENTED** [review]
Thanks for updating this! Looks like that was just overlooked in https://github.com/SciTools/cartopy/pull/2166

File: lib/cartopy/crs.py
Comment (e-k-m) [reply lib/cartopy/crs.py:1]: Yes, done.

File: lib/cartopy/tests/mpl/test_mpl_integration.py (hunk 1 above)
Comment (e-k-m) [reply lib/cartopy/tests/mpl/test_mpl_integration.py]: Yes, done. THX.

**Review by greglucas — APPROVED** [review]
